### ISSUE TITLE
[🔨 compilation issue] Cannot compile GPU varient.

### ISSUE THREAD

**Issue by zznewclear13** [mitsuba-renderer/mitsuba2#569]
## Summary

Cannot compile GPU varient `gpu_autodiff_spectral`.

## System configuration

- Platform: windows 10
- Compiler: MSVC 14.29.30037
- Python version: I tried 3.7.9, 3.8.10, 3.9.10, 3.10.2, non of them worked.
- CMake: 3.22.2
- CUDA: 11.6
- Optix: 7.4
- Mitsuba 2 version: latest
- Compiled variants:
    * `scalar_rgb`
    * `scalar_spectral`
    * `gpu_autodiff_spectral`
- Build setting: Debug and Release

## Description

Some of the error messages are:

```
D:\Github\mitsuba2\src\librender\scene_optix.inl(173,31): error C2039: 'overrideUsesMotionBlur': is not a member of 'OptixPipelineLinkOptions' (compiling source file D:\Github\mitsuba2\src\librender\scene.cpp)
C:\ProgramData\NVIDIA Corporation\OptiX SDK 7.4.0\include\optix_7_types.h(1726): message : see declaration of 'OptixPipelineLinkOptions' (compiling source file D:\Github\mitsuba2\src\librender\scene.cpp)
D:\Github\mitsuba2\src\librender\scene_optix.inl(45): message : while compiling class template member function 'void mitsuba::Scene<enoki::DiffArray<enoki::CUDAArray<float>>,mitsuba::Spectrum<enoki::DiffArray<enoki::CUDAArray<float>>,4>>::accel_init_gpu(const mitsuba::Properties &)' (compiling source file D:\Github\mitsuba2\src\librender\scene.cpp)

D:\Github\mitsuba2\src\librender\scene_optix.inl(274,35): error C2039: 'aabbs': is not a member of 'OptixBuildInputInstanceArray' (compiling source file D:\Github\mitsuba2\src\librender\scene.cpp)
C:\ProgramData\NVIDIA Corporation\OptiX SDK 7.4.0\include\optix_7_types.h(569): message : see declaration of 'OptixBuildInputInstanceArray' (compiling source file D:\Github\mitsuba2\src\librender\scene.cpp)
D:\Github\mitsuba2\src\librender\scene_optix.inl(236): message : while compiling class template member function 'void mitsuba::Scene<enoki::DiffArray<enoki::CUDAArray<float>>,mitsuba::Spectrum<enoki::DiffArray<enoki::CUDAArray<float>>,4>>::accel_parameters_changed_gpu(void)' (compiling source file D:\Github\mitsuba2\src\librender\scene.cpp)

D:\Github\mitsuba2\src\librender\shape.cpp(284,17): error C2039: 'aabbArray': is not a member of 'OptixBuildInput'
C:\ProgramData\NVIDIA Corporation\OptiX SDK 7.4.0\include\optix_7_types.h(606): message : see declaration of 'OptixBuildInput'
D:\Github\mitsuba2\src\librender\shape.cpp(281): message : while compiling class template member function 'void mitsuba::Shape<float,mitsuba::Color<float,3>>::optix_build_input(OptixBuildInput &) const'
```
and after several lines, it outputs
```
LINK : fatal error LNK1181: cannot open input file 'D:\Github\mitsuba2\build\src\librender\mitsuba-render-obj.dir\Debug\scene.obj'

LINK : fatal error LNK1181: cannot open input file '..\Debug\mitsuba-render.lib'

LINK : fatal error LNK1181: cannot open input file '..\librender\Debug\mitsuba-render.lib'
```

## Steps to reproduce

1. Clone repository through Github Desktop.
2. Modify `mitsuba.conf`, add `gpu_autodiff_spectral`.
3. Open CMake(cmake-gui), set source code path `D:/Github/mitsuba2` and build path `D:/Github/mitsuba2/build`.
4. Press Configure, set `MTS_USE_OPTIX_HEADERS` to 1.
5. Press Configure, set `MTS_OPTIX_PATH` to `C:\ProgramData\NVIDIA Corporation\OptiX SDK 7.4.`
6. Press Configure, set `CUDA_SDK_ROOT_DIR` the same as `CUDA_TOOLKIT_ROOT_DIR` (C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v11.6).
7. Set `CMAKE_CUDA_ARCHITEXTURES` to 75.
8. Press Configure, Generate and Open Project.
9. In Visual Studio 2019, click Build -> Build Solution.
10. Get error messages.

## Other information

I succeeded building mitsuba on my first attempt. But when I imported mitsuba in python, it output `ImportError: The 'mitsuba' native modules could not be imported.`. Therefore I tried to recompile mitsuba with different python versions. But ever since then, every build was failed.

**Comment by Speierers** [comment]
Hi @zznewclear13 ,

Optix 7.4 isn't supported yet. Could you try with OptiX 7.2. instead?

**Comment by zznewclear13** [comment]
Hi @Speierers ,

I made a quick test with Optix 7.2, CUDA 11.6, Python 3.8.7, CMake 3.18.3, `CMAKE_CUDA_ARCHITEXTURES ` : 52, and it failed again with following errors:

```
LINK : fatal error LNK1181: cannot open input file 'G:\myFolder\GraphicsCollection\mitsuba2\build\src\librender\mitsuba-render-obj.dir\Release\scene.obj'

LINK : fatal error LNK1181: cannot open input file '..\Release\mitsuba-render.lib'

LINK : fatal error LNK1181: cannot open input file '..\librender\Release\mitsuba-render.lib'
```

And I just checked my `mitsuba2\build\src\librender\mitsuba-render-obj.dir\Release\` path, it seemed that CMake did not create scene.obj.

I'll keep on trying with different CUDA versions.

**Comment by Speierers** [comment]
Make sure to build all the targets in Visual Studio

**Comment by zznewclear13** [comment]
Hi,

Finally I figured out my issue! Mitsuba has it's own optix header file `optix_api.h`, which is based on Optix 7.0.0. Later Optix versions define different structures than Optix 7.0.0. So when enabling `MTS_USE_OPTIX_HEADERS` with Optix later than 7.0.0, it will throw errors.

In case anyone wanted to know, I compiled `gpu_autodiff_spectral` with CUDA 11.6, Optix 7.0.0, Python 3.7.8, CMake 3.18.3, with `MTS_USE_OPTIX_HEADERS` enabled and `CMAKE_CUDA_ARCHITEXTURES` set to 52.

For anyone has issues with importing mitsuba python package, please check your system environment variables. You should have `PATH`, `PYTHONPATH`, and `MITSUBA_DIR` set to corresponding paths.

**Comment by Mephisto405** [comment]
First, your comments are so helpful @zznewclear13. Thanks.

However, I got another problem so I reopen the issue.

My system consists of RTX 3070, CUDA 11.2, OptiX 7.0.0, Python 3.8.8.
I compiled `gpu_autodiff_spectral` with `MTS_USE_OPTIX_HEADERS` enabled and `CMAKE_CUDA_ARCHITEXTURES` set to 86.

This can be done by the following command in the `build` directory (assuming that you already shot `mkdir build` and `cd build`).
```bash
cmake .. -DCMAKE_CUDA_ARCHITEXTURES=86 -DMTS_USE_OPTIX_HEADERS=1 -DMTS_OPTIX_PATH={some absolute path that contains the directory of /include/optix.h}
```

I successfully (?) build the Mitsuba, and tried running the following python code.
```
import mitsuba

print(mitsuba.variant())
mitsuba.set_variant('gpu_rgb')
print(mitsuba.variant())
```

It returns:
```
None
rt_check(): OptiX API error = 7804 (Library not found) in ../src/librender/optix_api.cpp:179.
```

Note that I enabled `gpu_rgb` and `scalar_rgb` and when I changed `mitsuba.set_variant('gpu_rgb')` to `mitsuba.set_variant('scalar_rgb')`, it works fine.
